captum/attr/_core/occlusion.py

```diff
@@ -0,0 +1,282 @@
+                            to these arguments.
+                            Default: None
+                ablations_per_eval (int, optional): Allows multiple occlusions
+                            to be processed simultaneously in one call to forward_fn.
+                            Each forward pass will contain a maximum of
+                            ablations_per_eval * #examples samples.
+                            For DataParallel models, each batch is split among the
+                            available devices, so evaluations on each available
+                            device contain at most
+                            (ablations_per_eval * #examples) / num_devices
+                            samples.
+                            Default: 1
+
+        Returns:
+                *tensor* or tuple of *tensors* of **attributions**:
+                - **attributions** (*tensor* or tuple of *tensors*):
+                            The attributions with respect to each input feature.
+                            Attributions will always be
+                            the same size as the provided inputs, with each value
+                            providing the attribution of the corresponding input index.
+                            If a single tensor is provided as inputs, a single tensor is
+                            returned. If a tuple is provided for inputs, a tuple of
+                            corresponding sized tensors is returned.
+
+
+        Examples::
+
+            >>> # SimpleClassifier takes a single input tensor of size Nx4x4,
+            >>> # and returns an Nx3 tensor of class probabilities.
+            >>> net = SimpleClassifier()
+            >>> # Generating random input with size 2 x 4 x 4
+            >>> input = torch.randn(2, 4, 4)
+            >>> # Defining FeatureAblation interpreter
+            >>> ablator = Occlusion(net)
+            >>> # Computes occlusion attribution, ablating each 1x3x3 patch,
+            >>> # shifting in each direction by the default of 1.
+            >>> attr = ablator.attribute(input, target=1, occlusion_shapes=(1,3,3))
+        """
+        formatted_inputs = _format_input(inputs)
+
+        # Formatting strides
+        if strides is None:
+            strides = tuple(1 for input in formatted_inputs)
+        if len(formatted_inputs) == 1 and not (
+            isinstance(strides, tuple) and len(strides) == 1
+        ):
+            strides = (strides,)
+        assert isinstance(strides, tuple) and len(strides) == len(
+            formatted_inputs
+        ), "Strides must be provided for each input tensor."
+
+        # Construct occlusion blocks
+        if not isinstance(occlusion_shapes[0], tuple):
+            occlusion_shapes = (occlusion_shapes,)
+        assert len(occlusion_shapes) == len(
+            formatted_inputs
+        ), "Must provide occlusion dimensions for each tensor."
```

Comment:
Although it is formatting, which is mostly in common, this is pretty specific to occlusion, so not sure if it'll be confusing to have it in common? The same might be true of strides, but I guess there could be more use cases of strides? I can also add it as a helper method in occlusion, what do you think?

Reply:
I think we can still move it there and put a comment saying that it is used by occlusion this could be used also by other image sliding window algorithms ? 
Maybe it would be even better to call, `silding_window_shapes` ? Will it be more understandable ?

Ultimately, we could move formatting from  `common.py` into `common_formatting.py` to keep it more clear and avoid common.py becoming too large.

Reply:
In general I think that it is good to describe that this is similar to sliding window because occlusion term is not very well known. When we search for it, google redirects to dentist's web sites :D